### ISSUE TITLE
5zig breaks skin/cape changing

### ISSUE THREAD

**Issue by boomboompower** [boomboompower/SkinChanger#1]
Reported by **EpicNewb**

---

Multiple instances in 5zig break our cape loading methods.
Class list with method referencing below

```- ResourceManager#loadPlayerResource(GameProfile)
- eu.the5zig.mod.util.IVariables#CapeCallback.parseImage(BufferedImage)
- eu.the5zig.mod.gui.GuiUploadDefaultCape.<init>
- eu.the5zig.mod.gui.ingame.resource.CapeResource.<init>
- eu.the5zig.mod.gui.ingame.resource.PlayerResource#getCapeResource
- eu.the5zig.mod.gui.ingame.resource.PlayerResource#setCapeResource(CapeResource)
- eu.the5zig.mod.chat.network.packets.PacketCapeSettings.<init>
```

One Transformer class found
```
Appears to change the getCapeLocation() method to require two Object methods

- eu.the5zig.mod.asm.transformers.PatchAbstractClientPlayer#PatchGetResourceLocation.visitCode
```

**Comment by boomboompower** [comment]
This is a full list of **all** referencing code. Code found _may_ or _may not_ have anything to do with our skin/cape changing issues
```ResourceManager
- eu.the5zig.mod.util.IVariables
- eu.the5zig.mod.chat.entity.Profile
- eu.the5zig.mod.config.ConfigNew
- eu.the5zig.mod.gui.GuiCapeSettings
- eu.the5zig.mod.chat.network.Protocol
- eu.the5zig.mod.manager.SkinManager
- eu.the5zig.mod.gui.GuiUploadDefaultCape
- eu.the5zig.mod.gui.ingame.resource.CapeResource
- eu.the5zig.mod.gui.ingame.resource.PlayerResource
- eu.the5zig.mod.chat.network.packets.PacketCapeSettings
- eu.the5zig.mod.asm.transformers.PatchAbsractClientPlayer.PatchGetResourceLocation
```

**Comment by GamerGlitz** [comment]
I need this to be fixed. I use 5Zig a lot, and I need either SkinChanger to be updated, or someone to make a modification to 5Zig.

**Comment by boomboompower** [comment]
> I need this to be fixed. I use 5Zig a lot, and I need either SkinChanger to be updated, or someone to make a modification to 5Zig.


~~I'm labelling this as a WontFix due to it not being active since 2017. The code here is likely different now than it was before.~~

See my comment below, this has now been patched and will work with never versions of the SkinChanger mod.

**Comment by boomboompower** [comment]
Code for 5zig is available online now https://github.com/5zig/The-5zig-Mod

On a separate note, using Mixin fixes the issue with 5zig, I have tested this in both a development and a production environment. I am satisfied that this issue is resolved for the V3.0.0 release.
